esmvalcore/esgf/_search.py

```diff
@@ -150,6 +155,10 @@ def esgf_search_files(facets):
 
 def select_by_time(files, timerange):
     """Select files containing data between a timerange."""
+    if '*' in timerange:
+        # TODO: support * combined with a period
+        return files
```

Comment:
I was just wondering if this changes the behavior of the function, but it probably just failed before when you used a wildcard. Can we remove the workaround with this change now? Are they removed in this PR?

Reply:
Yes, this change prevents the failure, which is slightly nicer than the crash we had before. The problem with the current code is that find_files and its local equivalent are not able to handle wildcards. How the code currently works is that it first finds all input files, then updates the timerange, and then searches again for input files with the timerange specified, instead of searching just once and then filtering the results. I had a go at improving this in 885d7181bc9c929a0a4ee7c455265f36fd45f00b, but that's quite far removed from the current state of the code. I'm doing this from IS-ENES3 funding and that ends in March, so I'm not sure if we have enough time to get something like that merged.